File: 0238-product-of-array-except-self/solution.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        solution = [1] * len(nums)
        prefix, postfix = (1,1)
        for i in range(len(nums)):
            solution[i] = prefix
            prefix =  prefix * nums[i]
        for i in range(len(nums) -1, -1, -1):
            solution[i] = postfix * solution[i]
            postfix = postfix * nums[i]
        return solution

File: 0238-product-of-array-except-self/solution.go
func productExceptSelf(nums []int) []int {
   result := make([]int, len(nums))
   for i, _ := range result {
    result[i] = 1
   }
   prefix := 1
   for i, _ := range nums {
    result[i] = prefix
    prefix *= nums[i]
   }
   postfix := 1
   for i := len(nums) - 1; i >= 0; i-- {
    result[i] *= postfix
    postfix *= nums[i]
   }
   return result
}
